util/generate_ticker_file.py

```diff
@@ -0,0 +1,173 @@
+                unique_list[i], original_entity))
+            possible_conflicts.append(index)
+        else:
+            unique_list[index] = ' '.join(
+                original_entity[:len(current_entity.split(' ')) + 1])
+
+    if len(possible_conflicts) > 1:
+        print(
+            "ERROR: More than one duplicate could not be resolved at entity {0} of {1}".
+            format(i + 1, len(original_list)))
+
+    return unique_list
+
+
+def update_entity_working_list(original_entities, seen_entities, entity_working_list, i):
+    if entity_working_list[i] in seen_entities:
+        first_duplicate = seen_entities.index(entity_working_list[i])
+        entity_working_list = patch_entity_list_duplicates(entity_working_list,
+                                                           original_entities, i)
+        seen_entities[first_duplicate] = entity_working_list[first_duplicate]
+
+    return seen_entities, entity_working_list
+
+
+def create_unique_minimum_entities(entities):
+    """
+    Retuns an entity list where each entity is as short
+    as possible while maintaining a unique list
+  
+    >>> create_unique_minimum_entities(['abc one company', 'abc two company', 'def three company'])
+    ['abc one', 'abc two', 'def']
+    """
+    original_entities = entities
+    unique_entity_working_list = [e.split(' ')[0] for e in original_entities]
+
+    while True:
+        seen_entities = []
+        i = 0
+        while i < len(unique_entity_working_list):
+            seen_entities, unique_entity_working_list = update_entity_working_list(
+                original_entities,
+                seen_entities,
+                unique_entity_working_list,
+                i,
+            )
+            seen_entities.append(unique_entity_working_list[i])
+            i += 1
+
+        if len(seen_entities) == len(unique_entity_working_list):
+            break
+
+    return unique_entity_working_list
+
+
+def entities_to_rockets(entity_pairs, case_insensitive=False):
+    """
+    Convert entity pairs to rocket replacements
+  
+    >>> entities_to_rockets([['Test','TST']], True)
+    ['test -> TST']
+  
+    >>> entities_to_rockets([['Test','TST']])
+    ['Test -> TST']
+    """
+    entities = [clean_entity(e[0]) if case_insensitive else e[0] for e in entity_pairs]
+    entities = create_unique_minimum_entities(entities)
+
+    tickers = [e[1].strip() for e in entity_pairs]
+
+    assert len(entities) == len(tickers)
+
+    return ["{0} -> {1}".format(x, y) for x, y in zip(entities, tickers)]
+
+
+def tickers_to_rockets(entity_pairs):
+    """
+    Converts tickers to rocket replacements
+  
+    >>> tickers_to_rockets([['Test','TST']])
+    ['t s t -> TST']
+    """
+    tickers = [e[1].strip() for e in entity_pairs]
+    entities = [' '.join(list(e.lower())) for e in tickers]
+
+    return ["{0} -> {1}".format(x, y) for x, y in zip(entities, tickers)]
+
+
+def generate_ticker_file(filename,
```

Comment:
```suggestion
def generate_ticker_file(filename,|
  """
  * Create a companies.csv file with two columns for company name and ticker: company name, ticker
  corp one, ABC
  corp two, DEV
  three, GHI
  
  * Generate a company -> ticker mapping:
  python util/generate_ticker_list.py companies.csv
  and you get a file like this called companies.txt:

  corp two -> DEV
  corp one -> ABC
  three -> GHI
  
  * Generate a ticker -> ticker mapping:
  python util/generate_ticker_file.py companies.csv --tickers_only and you'll get a file like this called companies_tickers.txt:

  d e v -> DEV
  a b c -> ABC
  g h i -> GHI
  
  If you wish to use both files for entity definitions, simply combine them.
  cat companies_tickers.txt >> companies.txt
  """
```

Reply:
Thanks -- this will go in the wiki too once merged.